libbeat/docs/newbeat.asciidoc

```diff
@@ -146,203 +214,222 @@ NOTE: To be consistent with other Beats, you should append `beat` to your Beat n
 Let's go through each of the methods in the `Beater` interface and look at a
 sample implementation.
 
-[[config-method]]
-==== Config Method
+[[new-function]]
+==== New function
 
-The `Config` method deals with the configuration file and optionally with
-custom CLI flags.
 
-The recommended way of handling the configuration is to create a
-`ConfigSettings` type that matches the structure of the expected configuration
-file. Here is an example configuration section for Topbeat:
+The `New()` function receives the configuration options defined for the Beat and
+creates a Beat object based on them. In our case the `Lsbeat` object has the following signature:
+
+[source,go]
+----------
+type Lsbeat struct {
+	done   chan struct{}
+	config config.Config
+	client publisher.Client
+}
+----------
+
+- `done`: Channel used by the `Run()` function to stop when the `Stop()` function is called.
+- `config`: Configuration options for the Beat
+- `client`: Publisher that takes care of sending the events to the
+  defined output.
```

Comment:
The client is also closed on `Stop`, so the disconnects from publisher pipeline, potentially unblocking a blocked publish (if pipeline is full). This point should be stressed somewhere, as it is crucial to proper shutdown (beat not hanging).